scripts/tier_analyzer.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+
+import argparse
+import re
+import subprocess
+import sys
+
+
+def check_virtualenv():
+    """Check if we are in virtualenv and if not, exit."""
+    if not hasattr(sys, 'real_prefix'):
+        print('You must activate CFME virtualenv in oder to run this script.')
+        sys.exit(1)
+
+
+def get_pytest_output(args, use_tier_marker=False):
+    """Get string output of pytest command."""
+    print('Getting pytest output {} tier marker.'.format('with' if use_tier_marker else 'without'))
+    pytest_args = ['pytest',
+        args.test_path,
+        '--collect-only',
+        '--long-running',
+        '--use-provider',
+        args.provider,
+        '-m']
+    if use_tier_marker:
+        pytest_args.append(args.tier_marker)
+    else:
+        pytest_args.append('uses_testgen')
+
+    return subprocess.check_output(pytest_args)
+
+
+def parse_pytest_output(output):
+    """Parse raw pytest output to get list of parametrized test cases."""
+    test_cases = []
+    tc_regex = re.compile(r"\s*<Function '(?P<test_case_name>test_\w+)(?P<parameters>\[.+\])?'>")
+    for line in output.splitlines():
+        match = re.match(pattern=tc_regex, string=line)
+        if match:
+            test_cases.append('{}{}'.format(
+                match.group('test_case_name'),
+                match.group('parameters') if match.group('parameters') else ''))
+
+    return test_cases
+
+
+def compare_parsed_outputs(all, tiers):
+    """Get sorted list of test cases that are not marked with tiers."""
+    diff = set(all).difference(set(tiers))
+    return sorted(diff)
+
+
+def print_message(args, diff, all_parsed, all_count, tiers_parsed, tiers_count):
+    """Human-readable output."""
+    def sep():
+        print('*' * 60)
+
+    if args.verbose:
+        sep()
+        print('There are {} test_cases run for {} provider:\n'.format(all_count, args.provider))
+        for test_case in all_parsed:
+            print(test_case)
+        sep()
+        print('There are {} test_cases marked with "{}":\n'.format(tiers_count, args.tier_marker))
+        for test_case in tiers_parsed:
+            print(test_case)
+
+    sep()
+    print('There are {} test cases not marked with "{}":\n'.format(len(diff), args.tier_marker))
+    for test_case in diff:
+        print(test_case)
+    sep()
+
+
+if __name__ == '__main__':
+
+    parser = argparse.ArgumentParser(description="""This simple script lists all tests generated
```

Comment:
@jamesooden There is some simple description. Is that sufficient?

Reply:
Well, my preference is that all files start with a docstring at the top to explain what they are there for.   This servers two purposes.   The first is, the first question that comes to mind when you open a file is answered, which is what is this thing?   The second is docstrings can be picked up as actual documentation, whereas your argparser description will not provide for that.